packages/cw-derive/examples/basic.rs

```diff
@@ -1,30 +1,41 @@
 #![allow(dead_code)]
-use cosmwasm_std::Response;
-
-pub struct Ctx;
-pub struct Error;
+use cosmwasm_std::{DepsMut, Env, MessageInfo, Response};
 
 #[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq, schemars::JsonSchema)]
 pub struct Member;
 
 #[cw_derive::interface(module=msg, exec=Execute, query=Query)]
 pub trait Cw4 {
+    type Error;
+
     #[msg(exec)]
-    fn update_admin(&self, ctx: Ctx, admin: Option<String>) -> Result<Response, Error>;
+    fn update_admin(
+        &self,
+        ctx: (DepsMut, Env, MessageInfo),
```

Comment:
Happy if you add a type for this tuple:

`pub type CtxMut = (DepsMut, Env, MessageInfo);`
`pub type Ctx = (Deps, Env);`

Fine for the shape of it, just seems like it would be useful for an alias. Or maybe there are too many variants and it doesn't make sense?

Actually, would be interesting to see how this works in practice when destructing as well...

Would you:

`let (deps, _ info) = ctx;` ?
Or `update_admin(&self, (deps, _info): CtxMut, admin: Option<String>)` ?
Or `ctx.0.storage` ?

Reply:
A mentioned, why not just the classic `deps`, `env` and `info`?